simpleviewinc/keg-cli
# Add parse config lib

**Ticket**: [VIS-1107](https://jira.simpleviewtools.com/browse/VIS-1107)

## Context
* To complete the `VIS-1107` ticket properly, we needed repo for managing config files
* So that's what this PR is for
* It has the added benefit of cleaning up the `keg-cli` due to migrating files to this new repo

## Goal
* Create a repo that allows loading templated config files
* This already exists in the keg-cli, but would be helpful in other repos
* So I extracted the logic out of the keg-cli src folder, and created a new repo
* Doing this also allowed me to fully remove the `src/libs/fileSys` lib from the keg-cli

## Updates
* Add `repos/parse-config`
  * This includes the `YML`, `ENV`, and `template` logic originally in `keg-cli/src`
  * Add more tests for better coverage then it had when in the `keg-cli`
* Updated `package.json` alias including the `jest-alias` config
* Removed `src/libs/fileSys`
  * `fileSys` logic was moved to `cli-utils`
    * The keg-cli was never updated to use the `cli-utils/fileSys` code due to the `YML` and `ENV` requirements
    * This update fixes this, and allow it to work with the `keg-cli` via `parse-config`
* Updated all `require` statements to use `cli-utils/fileSys` and `parse-config` where needed
  * **This and the new `parse-config` repo are the main reason for some many changed files**

## Testing
### Setup
* Pull this PR =>  `keg cli && keg pr 92`
* Install dependencies =>  `yarn install`
* Run cmd => `keg config open`
  * It should open your `~/.kegConfig` folder
  * Delete all files the `.tmp` folder if they exist
    * These are temp files only needed when starting a tap, so don't worry about removing them
  * Leave this editor open, because will use it for reference later

**Test 1**
* In a terminal start any tap.
  * I used `tap-snack` for these tests =>  `keg snack start`
* It should start the tap without issue
* Go back to the editor of the `~/.kegConfig` folder
  * Inspect the `.tmp` folder
  * There should be a yml config file for the tap you just started
    * [Looks like this](https://i.imgur.com/XFiE4d3.png)
  * Ensure the `service name` is correct
    * It should match the `service name` of default `docker-compose.yml` file in the taps `container` folder
    * In my case the service is `snack`, which you can see on `line 6` in the image linked above 
  * Ensure the labels were properly added to the file
    * There are a number of `traefik.<some-value>` labels used with `keg-proxy`
      * You know these are working if you can navigate to the tap in the browser
    * There are a number of `com.keg.env.<some-value>` labels
      * Most of them reference an ENV from the taps `values.yml` file
      * If the `com.keg.proxy.domain` is the correct value, you know these are working properly
        * The value should be the `subdomain` used when opening the tap in the browser
        * In my case
          * The URL is => `http://snack-tap-snack-ui.local.keghub.io/`
          * Which matches the `com.keg.proxy.domain` label value of `snack-tap-snack-ui`

**Test 2**
* Run the cmd `keg generate task --name test`
* It should ask you a number of questions
  * [Looks like this](https://i.imgur.com/TJcDQSO.png)
* Answer the questions either matching the answers from the image linked above, or enter your own
* Afterwards it should create a new task within the `keg-cli/src/tasks/<parent-task>`
* Open the task, and ensure the file and its content was correctly generated
  * [Looks like this](https://i.imgur.com/fW2rJsp.png)
